osu.Game/Overlays/BeatmapSet/LeaderboardModSelector.cs

```diff
@@ -0,0 +1,149 @@
+            AutoSizeAxes = Axes.Both;
+            InternalChild = modsContainer = new FillFlowContainer<ModButton>
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                AutoSizeAxes = Axes.Both,
+                Direction = FillDirection.Full,
+                Spacing = new Vector2(4),
+            };
+        }
+
+        private void selectionChanged(Mod mod, bool selected)
+        {
+            if (selected)
+                SelectedMods.Add(mod);
+            else
+                SelectedMods.Remove(mod);
+
+            if (!SelectedMods.Any() && !IsHovered)
+                highlightAll();
+        }
+
+        protected override bool OnHover(HoverEvent e)
+        {
+            if (!SelectedMods.Any())
+                modsContainer.Children.Where(button => !button.IsHovered).ForEach(button => button.Highlighted.Value = false);
+
+            return base.OnHover(e);
+        }
+
+        protected override void OnHoverLost(HoverLostEvent e)
+        {
+            base.OnHoverLost(e);
+
+            if (!SelectedMods.Any())
+                highlightAll();
+        }
+
+        public void DeselectAll() => modsContainer.ForEach(mod => mod.Selected.Value = false);
+
+        private void highlightAll() => modsContainer.ForEach(mod => mod.Highlighted.Value = true);
+
+        private class ModButton : ModIcon
+        {
+            private const float mod_scale = 0.4f;
+            private const int duration = 200;
+
+            public readonly BindableBool Selected = new BindableBool();
+            public Action<Mod, bool> OnSelectionChanged;
+
+            public ModButton(Mod mod)
+                : base(mod)
+            {
+                Scale = new Vector2(mod_scale);
+                Highlighted.Value = true;
+                Add(new HoverClickSounds());
+            }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                Selected.BindValueChanged(selected =>
+                {
+                    updateState();
+                    OnSelectionChanged?.Invoke(Mod, selected.NewValue);
+                });
+            }
+
+            protected override bool OnClick(ClickEvent e)
+            {
+                Selected.Toggle();
+                return true;
+            }
+
+            protected override bool OnHover(HoverEvent e)
+            {
+                updateState();
+                return false;
+            }
+
+            protected override void OnHoverLost(HoverLostEvent e)
+            {
+                base.OnHoverLost(e);
+                updateState();
+            }
+
+            private void updateState() => Highlighted.Value = IsHovered || Selected.Value;
+
+            protected override void OnHighlightedChanged(ValueChangedEvent<bool> highlighted) =>
+                this.FadeColour(highlighted.NewValue ? Color4.White : Color4.Gray, duration, Easing.OutQuint);
```

Comment:
the highlight and hover state should not be the same. it's really hard to tell you actually selected anything on clicking.

Reply:
How about make it behave like in mod overlay? I mean, scale it up on selection?